### DISCUSSION TITLE
Is there any way to get the previous voice state?

### DISCUSSION THREAD

**Discussion by kvizyx** [disgoorg/disgo#358] [Help] [answered]
I didn't quite put it right in the last question.

In my code i have a map of structs that represents "sessions" that should be identified for specific voice channel (for now identifier is channel id) so I know which of them to send an event to. When someone leave voice channel i need to send event about it to session that is linked to that channel so i can update number of channel members, and that's a problem because in leave event the only identifier that can be used to do this is the session id.

Is there any way to get the previous voice state or maybe session id from voice connection request so i can link my internal session to that session id?

**Comment by topi314** [comment, marked as answer]
Don't worry about following up in a resolved question.

Since disgo keeps track of all voice states if you enable the cache the voice state events carries a `OldVoiceState` field seee [here](https://pkg.go.dev/github.com/disgoorg/disgo@v0.18.5/events#GuildVoiceStateUpdate)

**Comment by kvizyx** [comment]
Thank you once again :)
